openedx/core/djangoapps/user_api/accounts/api.py

```diff
@@ -182,6 +199,12 @@ def update_account_settings(requesting_user, update, username=None):
         for serializer in user_serializer, legacy_profile_serializer:
             serializer.save()
 
+        if "account_privacy" in update:
+            if account_privacy is not None:
+                account_privacy_serializer.save()
+            else:
+                delete_user_preference(requesting_user, "account_privacy")
+
```

Comment:
In case of account update; there are two steps.
1. Validate; if there are any field errors, raise exception else,
2. Save
The `update_user_preferences` in `preferences/api.py` does both steps for preferences. 
If we call it directly then shall there exist any error in preference field `account_privacy`; it will be raised after update of other account information. That is why I explicitly validated and saved account_privacy in two steps.